rs/nns/governance/canister/canister.rs

```diff
@@ -378,9 +378,11 @@ fn list_neurons(req: ListNeurons) -> ListNeuronsResponse {
 }
 
 #[query]
-fn get_neuron_index(_req: GetNeuronIndexRequest) -> Result<NeuronIndexData, GovernanceError> {
+fn get_neuron_index(req: GetNeuronIndexRequest) -> Result<NeuronIndexData, GovernanceError> {
     debug_log("get_neuron_index");
-    Ok(NeuronIndexData::default())
+    governance()
+        .get_neuron_index(req, caller())
+        .map_err(GovernanceError::from)
```

Comment:
I think you can delete this line. Is it needed for reasons that I'm not seeing??

Reply:
OMG 🤦‍♂️

I truly loath these duplicate types. Maybe I can get used to them, but they just seem like a pervasive "trip hazard". At the very least, we can give them different names so that we can always tell what we are actually looking at. E.g. ApiGovernanceError vs. PbGovernanceError. I mean, imagine having (identical) twins, but they ALSO have the exact same name! That should probably be illegal (I would not be surprised if naming your children the same is already illegal).